### PR TITLE
fix: adds support for ipv6 addresses

### LINKED ISSUE #23: Support ipv6 clusters
When running cortex-tenant in an ipv6 cluster, I'm hitting this error:

```
time="2022-12-15T15:15:36Z" level=error msg="proc: src=127.0.0.1:45666 couldn't find DNS entries for the given domain. Try using DialDualStack"
```

Does cortex-tenant only support querying for A records and not AAAA?

### PR DESCRIPTION
**Pull request by ajpauwels**
- FastHTTP only queries A records by default; this adds support for AAAA records

fixes #23

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-1.7.2
+1.7.3
```
Comment (marcinfigiel) [inline VERSION:1]: Version 1.7.2 was not released yet, so this change is potentially unnecessary, see #22

File: config.go [hunk 2]
```diff
@@ -14,7 +14,8 @@ type config struct {
 	Listen      string
 	ListenPprof string `yaml:"listen_pprof"`
 
-	Target string
+	Target      string
+	SupportIPv6 bool `yaml:"support_ipv6"`
```
Comment (marcinfigiel) [inline config.go:18]: Arguably `enable` would fit better than `support`, i.e.:
```suggestion
	EnableIPv6 bool `yaml:"enable_ipv6"`
```

File: config.yml [hunk 3]
```diff
@@ -2,6 +2,7 @@ listen: 0.0.0.0:8080
 listen_pprof: 0.0.0.0:7008
 
 target: http://127.0.0.1:9091/receive
+support_ipv6: false
```
Comment (marcinfigiel) [inline config.yml:5]:
```suggestion
enable_ipv6: false
```

File: processor.go [hunk 4]
```diff
@@ -65,6 +65,7 @@ func newProcessor(c config) *processor {
 		WriteTimeout:       c.Timeout,
 		MaxConnWaitTimeout: 1 * time.Second,
 		MaxConnsPerHost:    64,
+		DialDualStack:      c.SupportIPv6,
```
Comment (marcinfigiel) [inline processor.go:68]:
```suggestion
		DialDualStack:      c.EnableIPv6,
```

**Review by marcinfigiel — CHANGES_REQUESTED** [review]
Please consider my inline suggestions.
Beside that the new configuration option is missing from the README.md file (there is a configuration example there which contains and explains all the options).

File: VERSION (hunk 1 above)
Comment (ajpauwels) [reply VERSION]: Reverted

File: config.go (hunk 2 above)
Comment (ajpauwels) [reply config.go]: Changed name to `EnableIPv6` and `enable_ipv6`

File: processor.go (hunk 4 above)
Comment (ajpauwels) [reply processor.go]: Changed name to `EnableIPv6` and `enable_ipv6`

**Review by marcinfigiel — APPROVED** [review]
Cool, thank you! Now let's hope @blind-oracle merges it 😃

**Comment by ajpauwels** [comment]
Sweet! This will be invaluable as a dual- and single-stack IPv6 clusters become more common :)

**Comment by blind-oracle** [comment]
Thanks for the fix! Maybe we should just enable it by default and don't add another option? Probably it should not break anything

**Comment by marcinfigiel** [comment]
I wouldn't change the default, because it alters the behavior a bit and might potentially break something for someone already running the software. FastHTTP leaves it disabled for a reason: https://github.com/valyala/fasthttp/blob/v1.43.0/client.go#L203

**Comment by blind-oracle** [comment]
Well, in private networks it shouldn't probably break, but it makes sense, I agree.

**Pull request merged**